### PR TITLE
Add support for Youtube and Twitter links rewrite

### PR DESCRIPTION
**Pull request by PrasannaVenkadesh**
Based on feature request, #1710

### CODE REVIEW

**Review by 0xhtml — APPROVED** [review]

**Comment by dalf** [comment]
It would be nice to update the thumbnail and embedded fields too, so no requests are sent to youtube / twitter.

https://github.com/asciimoo/searx/blob/master/searx/engines/youtube_noapi.py#L32


```python
...
embedded_url = '<iframe width="540" height="304" ' +\
    'data-src="https://www.youtube-nocookie.com/embed/{videoid}" ' +\
    'frameborder="0" allowfullscreen></iframe>'
...
thumbnail = 'https://i.ytimg.com/vi/' + videoid + '/hqdefault.jpg'
```

No sure it is in the scope of these plugins, but it would be even better if the thumbnail and embedded would be added automaticaly.
Use case: some time a general purpose engine (or google video) may returns results from youtube / twitter. In this case the thumbnail / embedded are not provided.
This plugins (or somewhere else) could :
* add the thumbnail and embedded fields.
* change to template field to videos.html

**Comment by PrasannaVenkadesh** [comment]
@dalf Thanks for the suggestion. Let me see it.

**Comment by Perflyst** [comment]
Your template options in settings.yml say that the plugin should be enabled in the admin web-interface. How can I enable it through settings.yml?

File: searx/plugins/twitter_rewrite.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+searx is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with searx. If not, see < http://www.gnu.org/licenses/ >.
+
+(C) 2015 by Adam Tauber, <asciimoo@gmail.com>
+'''
```
Comment (return42) [inline searx/plugins/twitter_rewrite.py:16]: It is your implementation, you hold the copyright, not Adam Tauber!

- Copyright is the legal term used to declare and prove who owns the intellectual property (the code, text, etc.).
- Licensing is the legal term used to describe the terms under which people are allowed to use the copyrighted material.

@asciimoo what do you think .. should we switch to [SPDX License Identifier](https://spdx.org/using-spdx-license-identifier)? .. see issue  #1801

Comment (asciimoo) [reply searx/plugins/twitter_rewrite.py:16]: Yeah, SPDX is good for me.

**Comment by sevospl** [comment]
@asciimoo It's been already months. Why is it still not merged? I don't get it.
All checks have passed. It's such a great feature worth using. It'd definitely increase our privacy.

**Comment by return42** [comment]
Is it only me; `make test`spits out an error:

```bash
$ make test
PYENV     usage: source ./local/py3/bin/activate
PYENV     using virtualenv from ./local/py3
PYENV     install .
. ./local/py3/bin/activate; ./manage.sh pep8_check
[!] Running pep8 check
/x/searx/local/py3/lib/python3.7/site-packages/pep8.py:109: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
. ./local/py3/bin/activate; ./manage.sh unit_tests
[!] Running unit tests
.........................................................................
----------------------------------------------------------------------
Ran 73 tests in 0.259s

OK
. ./local/py3/bin/activate; ./manage.sh install_geckodriver
[!] Checking geckodriver
. ./local/py3/bin/activate; ./manage.sh robot_tests
[!] Running robot tests
DEBUG:searx:read configuration from /x/test123/searx/searx/settings_robot.yml
INFO:searx:Initialisation done

Running 7 tests
Error occured: Traceback (most recent call last):
  File "/x/searx/searx/testing.py", line 107, in <module>
    run_robot_tests([getattr(robot, x) for x in dir(robot) if x.startswith('test_')])
  File "/x/searx/searx/testing.py", line 75, in run_robot_tests
    test(browser)
  File "/x/searx/tests/robot/__init__.py", line 16, in test_404
    browser.visit(url + 'missing_link')
  File "/x/searx/local/py3/lib/python3.7/site-packages/splinter/driver/webdriver/__init__.py", line 212, in visit
    self.driver.get(url)
  File "/x/searx/local/py3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 333, in get
    self.execute(Command.GET, {'url': url})
  File "/x/searx/local/py3/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/x/searx/local/py3/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: Reached error page: about:neterror?e=connectionFailure&u=http%3A//localhost%3A11111/missing_link&c=UTF-8&f=regular&d=Firefox%20kann%20keine%20Verbindung%20zu%20dem%20Server%20unter%20localhost%3A11111%20aufbauen.

```

**Comment by return42** [comment]
FYI: after https://github.com/asciimoo/searx/pull/1938 gets merged, this could be implemented as external plugin.

Since this plugin need a bugfix and I got no feedback, I close this PR.  Ask, if you want to reopen this PR / Thanks!

**Pull request closed without merging**